### ISSUE TITLE
Toolbar to be able to reset to default when the controller is cleared

### ISSUE THREAD

**Issue by alesalv** [fleather-editor/fleather#383]
**Is your feature request related to a problem? Please describe.**
Related to [this](https://github.com/fleather-editor/fleather/issues/382), at the moment when I have:

```
    return Column(
      children: [
        FleatherToolbar.basic(
            controller: controller),
        FleatherEditor(
            controller: controller),
    );
```
I edit my message, I highlight some text and add a list, so I click the toolbar list icon and then type 3 lines, then I send the message, and I clean up the editor with this [workaround](https://github.com/fleather-editor/fleather/issues/265#issuecomment-1925615845), what happens is that the text is cleared but the list functionality of the toolbar is still selected, so you see:

`1. (cursor)
`

instead of a fully empty text.

**Describe the solution you'd like**
I would like that when the editor is cleared, for instance with the newly introduced controller.clear(), the toolbar would listen to it and reset itself as well from any clicked icon I selected while editing

**Describe alternatives you've considered**
I try to reset the toolbar using some callback, but I couldn't, only way was to redraw a new one completely

**Comment by Amir-P** [comment]
This is because the last newline of the document in your case has list attribute and in the workaround it doesn't get unset. I'm gonna close this since the `controller.clear` method we want to add should take care of this too.

**Comment by alesalv** [comment]
@Amir-P technically you should leave this improvement open, and close it only once the other improvement is released 😸 
Until then, anything I can do as a workaround or hack, to reset the toolbar?

**Comment by Amir-P** [comment]
I decided to close the issue since it's not related to Fleather. It's about a workaround I mentioned in another issue and should be fixed after fixing #382.
Anyway, I guess this should fix your problem for now, but let me know if you encounter any issues. @alesalv 

```dart
final documentLength = _controller!.document.length;
_controller!
    .replaceText(0, 1, '\n', selection: TextSelection.collapsed(offset: 0));
_controller!.replaceText(1, documentLength, '');
```

**Comment by alesalv** [comment]
Thanks, I'll try this workaround in a few.
The part of this issue I think it's related to Fleather is in the FleatherToolbar, as while the controller manipulates its context, the toolbar doesn't listen to it. But for sure everything will be fixed once the reset() is introduced
